server/lib/nl/fulfillment/triple.py

```diff
@@ -23,6 +23,42 @@
 # Handler for TRIPLE queries.
 #
 
+_OUT_ARROW = '->'
+_IN_ARROW = '<-'
+
+
+# Gets the chart vars for a property and direction if values exist for that
+# property and direction,
+def _get_chart_vars(uttr: nl_uttr.Utterance, prop: str, out: bool) -> ChartVars:
+  entity_dcids = [e.dcid for e in uttr.entities]
+  prop_values = raw_property_values(entity_dcids, prop, out)
+  if not any(prop_values.values()):
+    return None
+  prop_with_direction = f'{_OUT_ARROW}{prop}' if out else f'{_IN_ARROW}{prop}'
+  chart_vars = ChartVars(props=[prop_with_direction])
+  return chart_vars
+
+
+# Handles properties with no direction specified
+def _populate_any_direction_prop(uttr: nl_uttr.Utterance, prop: str):
+  out_chart_vars = _get_chart_vars(uttr, prop, True)
+  in_chart_vars = _get_chart_vars(uttr, prop, False)
```

Comment:
TODO:  Try to do concurrent calls / prefer index to use properties with arrows?

Reply:
done